### DISCUSSION TITLE
Help understanding a misbehaving rule in the php parser

### DISCUSSION THREAD

**Discussion by cfroystad** [tree-sitter/tree-sitter#1140] [Q&A] [answered]
Hi Community 🙂 

Thank you all for being part of this interesting and great project - I've learned so much by participating here!

Today, I've come to ask your assistance with a problem I'm facing while bringing tree-sitters PHP support up to speed. 

While working on parsing the content of strings, I've gotten stuck on the following rule:
```
_simple_string_part: $ => prec.right(choice(
  alias($._simple_string_member_access_expression, $.member_access_expression),
  $._variable_name,
  alias($._simple_string_subscript_expression, $.subscript_expression),
)),
```
[See context here](https://github.com/tree-sitter/tree-sitter-php/pull/72/files#diff-919ac210accac9ecc55a76d10a7590e3d85ca3f0e165b52d30f08faee486d0cbR1184)

The test data I use:
```
"$people->john drank some $juices[0] juice.".PHP_EOL;
"$people->john then said hello to $people->jane.".PHP_EOL;
"$people->john's wife greeted $people->robert.";
"The character at index -2 is $string[-2].";
```

$people, $string and $juice are all variables, while $people->john is an example of member access and $juices[0] is an example of subscript (I'm sure you know this better than me, but in case you're not familiar with PHP).

The strange thing is that if we leave the _simple_string_part rule as is, all variables are matched as variables and the member and subscript parts are interpreted as strings. However, if I remove the $._variable_name rule, the member and subscript expressions are correctly matched (and all sorts of errors across the test suite occurs for very good reasons).

In short: How to make _simple_string_part choose the right node and not always choose $._variable_name?

I'm guessing it's a precedence problem, but I've yet to get that far in my tree-sitter journey to be able to figure it out and would be very grateful for a pointer or two in the right direction!

**Comment by cfroystad** [comment, marked as answer]
Solved in https://github.com/tree-sitter/tree-sitter/discussions/1339
